lib/buffer.js

```diff
@@ -322,13 +322,31 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  if (start >= this.length || end <= 0)
+    return '';
+
+  /*
+   * if unsigned 32 bit value (>>> 0) of `start` is not the same as `start`,
+   * use `0` as the default value. Similarly, if the unsigned 32 bit value of
+   * `end` is not the same as `end`, then use the length of the buffer as the
+   * end value.
+   *
+   * Note that, the comparison is done with abstract equality operator (!=) to
+   * allow stringified numbers ('1') and Number objects (new Number(5)).
+   */
+  if (start < 0 || start >>> 0 != parseInt(+start))
```

Comment:
but may also want to note that this is technically an int32 check, not uint32, b/c the value will be coerced to a uint32 below with `start >>>= 0`. so we're saving ourselves an operation.